src/integrationtests/java/com/aws/greengrass/integrationtests/BridgeTest.java

```diff
@@ -166,4 +166,82 @@ void GIVEN_mqtt5_and_mapping_between_local_and_iotcore_WHEN_local_message_receiv
 
         subscribeCallback.getLeft().get(AWAIT_TIMEOUT_SECONDS, TimeUnit.SECONDS);
     }
+
+    @TestWithMqtt5Broker
+    @WithKernel("mqtt5_local_to_iotcore_retain.yaml")
+    void GIVEN_mqtt5_and_mapping_between_local_and_iotcore_with_retainAsPublished_WHEN_message_published_THEN_message_bridged_with_retain_flag(Broker broker)
+            throws Exception {
+        MqttMessage expectedMessage = MqttMessage.builder()
+                .topic("topic/toIotCore")
+                .payload("message".getBytes(StandardCharsets.UTF_8))
+                // mqtt5-specific fields below.
+                .userProperties(Collections.singletonList(new UserProperty("key", "val")))
+                .responseTopic("response topic")
+                .messageExpiryIntervalSeconds(1234L)
+                .payloadFormat(Publish.PayloadFormatIndicator.UTF8)
+                .contentType("contentType")
+                .retain(true)
+                .build();
+
+        Pair<CompletableFuture<Void>, Consumer<Publish>> subscribeCallback
+                = asyncAssertOnConsumer(p -> assertEquals(expectedMessage, MqttMessage.fromSpoolerV5Model(p)));
```

Comment:
can we subscribe/verify on local topic?  iot core doesn't support retain as published, so subscribing to iot core topic won't give us a message w/ retain value that we expect